eng/pipelines/templates/jobs/archetype-sdk-tests.yml

```diff
@@ -127,6 +127,7 @@ jobs:
   - template: /eng/pipelines/templates/steps/cmake-build.yml
     parameters:
       GenerateArgs: $(CmakeArgs)
+      BuildArgs: "$(BuildArgs)"
```

Comment:
@danieljurek - I am not the expert here, can you review/speak to this?

Reply:
This was not causing Main to be broken, but it was missing to add the build args from the test matrix ( -v and --parallel )

Reply:
This looks good to me. In the case of the Ubuntu and Windows 2019 machines, the VMs have 2 cores. For I/O bound operations, like building, the process can complete faster with a degree of parallelism that is greater than the number of cores. 

In the case of `cmake-build.yml` this parameter is inserted into the build command line: https://github.com/Azure/azure-sdk-for-cpp/blob/main/eng/pipelines/templates/steps/cmake-build.yml#L20

`BuildArgs` is a variable and its value is set in the matrix above (if the value is not set somewhere then it is an empty string). 

The template syntax: 
```
 cmake --build . ${{ parameters.BuildArgs }}
```
After template substitution becomes: 
```
 cmake --build . $(BuildArgs)
```
In the case of `MacOS_x64_with_unit_test` in the matrix: 
```
 cmake --build . -j 4
```

According to [cmake documentation for the `-j` parameter](https://cmake.org/cmake/help/v3.13/manual/cmake.1.html#build-tool-mode): 

> The CMAKE_BUILD_PARALLEL_LEVEL environment variable, if set, specifies a default parallel level when this option is not given.

Another way to get this outcome is to use the `CMAKE_BUILD_PARALLEL_LEVEL`. 

Either approach will get the same effects. It might be easier to use the command line approach taken in this PR as, if the command line is logged, one can copy and paste the logged command line locally. Otherwise one would have to dig through and reassemble the state of the environment variables at build time starting at the matrix and working up to the job level (in the event that there are some global variables defined)